app-info/generate-eos-extra

```diff
@@ -77,7 +77,8 @@ def main():
             raise ValueError(f"No AppStream ID for {flatpak_ref}")
 
         # FIXME: this is wrong, but without it most of the apps don't appear!
-        if appstream_id.endswith(".desktop"):
+        if (appstream_id.endswith(".desktop") and
+                len(appstream_id.split('.')) > 3):
```

Comment:
This is quite error prone (as is the original code), maybe we should add a map of ids in `eos-extra.txt` and their actual appstream ids and if something is not found in this map it just removes the suffix?

Reply:
With this patch, the check now follows the [AppStream spec](https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-id-generic) for app ids, creating a new map file would be a massive overkill for this case.

Reply:
If you look at the original code, I went to great lengths to map the Flatpak app ID to the appstream ID out of the Flatpak appstream, only to discover that it doesn't work for unknown reasons.

Reply:
Ok, I see, so according to the spec ids should always contain 3 "fields", although I'm not sure if that is enforced... Anyway, the changes lgtm then, but I would probably remove the `FIXME` and add a comment with the link to the spec instead then.

As for the map I mentioned, I was thinking of a simple map in the code itself with only `org.telegram.desktop` -> `org.telegram.desktop` and if not found in the map always remove the `.desktop` suffix, but yeah, your solution seems correct indeed.